packages/synapse-react-client/src/components/ColumnFilter.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react'
+import { useQueryContext } from './QueryContext/QueryContext'
+import Select, {
+  components,
+  ControlProps,
+  GroupBase,
+  MultiValue,
+} from 'react-select'
+import {
+  ColumnMultiValueFunction,
+  ColumnMultiValueFunctionQueryFilter,
+  FacetColumnResultValues,
+  QueryFilter,
+} from '@sage-bionetworks/synapse-types'
+import { useQuery } from '@tanstack/react-query'
+
+export type FilterProps = {
+  filterColumnName?: string
+}
+
+export const Control = ({
+  children,
+  ...rest
+}: ControlProps<any, boolean, GroupBase<any>>) => {
+  return (
+    <components.Control {...rest} className="form-control">
+      {children}
+    </components.Control>
+  )
+}
+
+const ColumnFilter: React.FC<FilterProps> = props => {
+  const queryContext = useQueryContext()
+  const {
+    executeQueryRequest,
+    queryMetadataQueryOptions,
+    getCurrentQueryRequest,
+  } = queryContext
+  const { data: queryMetadata } = useQuery(queryMetadataQueryOptions)
+  const [selectedTypes, setSelectedTypes] = useState<string[]>([])
```

Comment:
Instead of storing the selection in state, we can make the source of truth the current query and eliminate this state variable.

You can do something like
```
const selectedFacet = getCorrespondingSelectedFacet({columnName}, currentQuery.query.selectedFacets)
const selectedValues = (selectedFacet as FacetColumnValuesRequest).facetValues
```

`getCorrespondingSelectedFacet` is a utility function you can import.